src/renderers/shared/utils/ReactErrorUtils.js

```diff
@@ -22,14 +22,14 @@ var caughtError = null;
  * @param {*} a First argument
  * @param {*} b Second argument
  */
-function invokeGuardedCallback<A, B, C>(
+function invokeGuardedCallback<A, B>(
   name: string,
-  func: (a: A, b: B) => C,
+  func: (a: A, b: B) => void,
   a: A,
   b: B,
```

Comment:
Couple other things… We never pass a `b` arg, we could probably drop that entirely and make things even simpler.

`func` is an event handler most times but we also call it once with a pre-bound componentWillUnmount. I was guessing that was the problem (perhaps other flowified app code runs through that path which triggered the failure internally?)